statistics/row_sampler.go

```diff
@@ -234,17 +248,19 @@ func (s *baseCollector) collectColumnGroups(sc *stmtctx.StatementContext, cols [
 	colLen := len(cols)
 	datumBuffer := make([]types.Datum, 0, len(cols))
 	for i, group := range colGroups {
+		if len(group) == 1 {
+			// For the sinlge-column group, its FMSketch is the same as that of the corresponding column. Hence, we
+			// don't need to maintain its FMSketch. We just copy the corresponding column's FMSketch after iterating
+			// all rows. Also, we can directly copy TotalSize and NullCount.
+			continue
+		}
+		// We don't maintain the null counts information for the multi-column group.
 		datumBuffer = datumBuffer[:0]
-		hasNull := true
 		for _, c := range group {
 			datumBuffer = append(datumBuffer, cols[c])
-			hasNull = hasNull && cols[c].IsNull()
-			s.TotalSizes[colLen+i] += sizes[c] - 1
```

Comment:
If the column values is null, we should not add `sizes[c] - 1`(actually it is `-1`) to `s.TotalSizes[colLen+i]`. Here is the tikv side code:
https://github.com/tikv/tikv/blob/master/src/coprocessor/statistics/analyze.rs#L529-L542